File: bindings/python/examples/how_tos/accounts_and_addresses/list_addresses.py
import os

from dotenv import load_dotenv

from iota_sdk import Wallet

# This example lists all addresses in the account.

# This example uses secrets in environment variables for simplicity which
# should not be done in production.
load_dotenv()

wallet = Wallet(os.environ['WALLET_DB_PATH'])

address = wallet.address()

for address in addresses:
    print(address.address)

File: bindings/nodejs/examples/how_tos/accounts_and_addresses/list_addresses.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger } from '@iota/sdk';

// Run with command:
// yarn run-example ./how_tos/accounts_and_addresses/list_addresses.ts

// This example lists all addresses in the account
async function run() {
    initLogger();
    try {
        const wallet = new Wallet({
            storagePath: process.env.WALLET_DB_PATH,
        });

        const account = await wallet.getAccount('Alice');

        const addresses = await account.addresses();

        for (const address of addresses) console.log(address.address);
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
